src/main/java/dev/breeze/settlements/config/ConfigField.java

```diff
@@ -0,0 +1,140 @@
+package dev.breeze.settlements.config;
+
+import com.google.common.base.Preconditions;
+import dev.breeze.settlements.utils.itemstack.ItemStackBuilder;
+import lombok.Getter;
+import lombok.Setter;
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class representing a field in a config file
+ */
+@Getter
+public final class ConfigField<T> {
+
+    /**
+     * The config file wrapper that contains this field
+     */
+    @Nonnull
+    private final ConfigFileWrapper file;
+
+    /**
+     * Matching ConfigType value, used to ensure that this field is type-valid
+     */
+    @Nonnull
+    private final ConfigType type;
+
+    /**
+     * The name of the configuration field.
+     */
+    @Nonnull
+    private final String name;
+
+    /**
+     * Description of the configuration field, often located on top as a comment
+     */
+    @Nonnull
+    private final List<String> description;
+
+    /**
+     * The default value of this field
+     */
+    @Nonnull
+    private final T defaultValue;
+
+    /**
+     * The current value of the configuration field
+     */
+    @Nullable
+    private T value;
+
+    /*
+     * Config GUI fields
+     */
+    /**
+     * The itemstack to represent this field
+     * - defaults to a piece of paper
+     */
+    @Setter
+    @Nonnull
+    private ItemStack guiItem = new ItemStackBuilder(Material.PAPER).build();
+
+    public ConfigField(@Nonnull ConfigFileWrapper file, @Nonnull ConfigType type, @Nonnull String name, @Nonnull List<String> description,
+                       @Nonnull T defaultValue) {
```

Comment:
Really just a nitpick, but maybe use @ParametersAreNonnullByDefault

Reply:
> Looks pretty much good, did not fully expect you to make a full wrapper system for yaml xd, would you want to make a general config as well? With a debug option that way it can enable or disable on startup.

Yep, sounds like a good idea. Added in [this commit](https://github.com/breeze-devs/Settlements/pull/2/commits/1c6c7180c0bc2e239efd25c92a5a0643c7d0c04e).